zerver/tests/test_subs.py

```diff
@@ -1294,33 +1294,31 @@ def test_create_stream_policy_setting(self) -> None:
 
         # Cannot create stream because not an admin.
         stream_name = ['admins_only']
-        result = self.common_subscribe_to_streams(user_profile, stream_name)
+        result = self.common_subscribe_to_streams(user_profile, stream_name, allow_fail=True)
         self.assert_json_error(result, 'User cannot create streams.')
 
         # Make current user an admin.
         do_change_user_role(user_profile, UserProfile.ROLE_REALM_ADMINISTRATOR)
 
         # Can successfully create stream as user is now an admin.
         stream_name = ['admins_only']
-        result = self.common_subscribe_to_streams(user_profile, stream_name)
-        self.assert_json_success(result)
+        self.common_subscribe_to_streams(user_profile, stream_name)
 
         # Allow users older than the waiting period to create streams.
         do_set_realm_property(user_profile.realm, 'create_stream_policy',
                               Realm.POLICY_FULL_MEMBERS_ONLY)
 
         # Can successfully create stream despite being under waiting period because user is admin.
         stream_name = ['waiting_period_as_admin']
-        result = self.common_subscribe_to_streams(user_profile, stream_name)
-        self.assert_json_success(result)
+        self.common_subscribe_to_streams(user_profile, stream_name)
 
         # Make current user no longer an admin.
         do_change_user_role(user_profile, UserProfile.ROLE_MEMBER)
 
         # Cannot create stream because user is not an admin and is not older than the waiting
         # period.
         stream_name = ['waiting_period']
-        result = self.common_subscribe_to_streams(user_profile, stream_name)
```

Comment:
Removing `result=` here makes the assertion 2 lines down wrong; I think this test should store `result` on every call.

Reply:
fixed